pkg/auth/disabled_authhandler.go

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package auth
+
+import (
+	"time"
+
+	"github.com/cilium/cilium/pkg/auth/certs"
+	"github.com/cilium/cilium/pkg/policy"
+)
+
+// disabledAuthHandler implements an authHandler for auth type disable by just authenticate every request.
+type disabledAuthHandler struct {
+}
+
+func (r *disabledAuthHandler) authenticate(authReq *authRequest) (*authResponse, error) {
+	// Authentication trivially done
+	log.Debugf("auth: Successfully authenticated request")
+
+	return &authResponse{
+		expirationTime: time.Now().Add(1 * time.Minute),
+	}, nil
+}
+
+func (r *disabledAuthHandler) authType() policy.AuthType {
+	return policy.AuthTypeDisabled
```

Comment:
we won't be able to use an auth type from the constant. we have to return something like `policy.AuthType(100)` :thinking: